### PR TITLE
move software version checker config in toml config file

### PR DESCRIPTION
**Pull request by miiu96**
Move software version checker config parameters in config.toml file

### CODE REVIEW

File: core/statistics/softwareVersion/softwareVersion.go [hunk 1]
```diff
@@ -21,56 +18,76 @@ type SoftwareVersionChecker struct {
...
 }
 
 // StartCheckSoftwareVersion will check on a specific interval if a new software version is available
 func (svc *SoftwareVersionChecker) StartCheckSoftwareVersion() {
-	go func() {
-		for {
-			svc.readLatestStableVersion()
-			time.Sleep(svc.checkRandInterval)
+	var ctx context.Context
+	ctx, svc.cancelFunc = context.WithCancel(context.Background())
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/statistics/softwareVersion/softwareVersion.go:56]: done.

File: core/statistics/softwareVersion/softwareVersion.go [hunk 2]
```diff
@@ -21,56 +18,76 @@ type SoftwareVersionChecker struct {
...
+			return
+		case <-time.After(svc.checkInterval):
 		}
-	}()
+	}
+}
+
+// Close will close the endless running go routine
+func (svc *SoftwareVersionChecker) Close() error {
+	if svc.cancelFunc != nil {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/statistics/softwareVersion/softwareVersion.go:76]: done.

File: vm/systemSmartContracts/staking.go [hunk 3]
```diff
@@ -597,26 +597,26 @@ func (r *stakingSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
...
 
 	currentNonce := r.eei.BlockChainHook().CurrentNonce()
 	if currentNonce-registrationData.UnStakedNonce < r.unBondPeriod {
-		r.eei.AddReturnMessage("unBond is not possible for address because unBond period did not pass")
+		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for address %s because unBond period did not pass", string(args.Arguments[0])))
 		return vmcommon.UserError
 	}
 	if registrationData.JailedRound != math.MaxUint64 {
-		r.eei.AddReturnMessage("unBond is not possible for jailed nodes")
+		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for jailed key %s", args.Arguments[0]))
```
Comment (bogdan-rosianu) [inline vm/systemSmartContracts/staking.go:615]: `string(args.Arguments[0])` ?

File: vm/systemSmartContracts/staking.go [hunk 4]
```diff
@@ -597,26 +597,26 @@ func (r *stakingSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		return vmcommon.UserError
 	}
 	if len(registrationData.RewardAddress) == 0 {
-		r.eei.AddReturnMessage("cannot unBond a key that is not registered")
+		r.eei.AddReturnMessage(fmt.Sprintf("cannot unBond key %s that is not registered", string(args.Arguments[0])))
```
Comment (bogdan-rosianu) [inline vm/systemSmartContracts/staking.go:600]: you can define a variable `blsKey := string(args.Arguments[0])` and use it

File: core/errors.go [hunk 5]
```diff
@@ -46,6 +46,9 @@ var ErrNilAppStatusHandler = errors.New("appStatusHandler is nil")
 // ErrNilStatusTagProvider signals that a nil status tag provider has been given as parameter
 var ErrNilStatusTagProvider = errors.New("nil status tag provider")
 
+// ErrInvalidPollingInterval signals that a invalid polling interval has been provided
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/errors.go:49]: done.

File: vm/systemSmartContracts/staking.go (hunk 3 above)
Comment (miiu96) [reply vm/systemSmartContracts/staking.go:615]: done.

File: vm/systemSmartContracts/staking.go (hunk 4 above)
Comment (miiu96) [reply vm/systemSmartContracts/staking.go:600]: i will let as it is

File: vm/systemSmartContracts/staking.go [hunk 6]
```diff
@@ -612,7 +612,7 @@ func (r *stakingSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		return vmcommon.UserError
 	}
 	if registrationData.JailedRound != math.MaxUint64 {
-		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for jailed key %s", args.Arguments[0]))
+		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for jailed key %s", string(args.Arguments[0])))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/staking.go:615]: hex.EncodeToString as the argument provided are raw bytes

**Pull request merged**